src/app/home/home.page.ts

```diff
@@ -47,43 +47,56 @@ export class HomePage {
   constructor(private countryService: CountryService) { }
 
   ngAfterViewInit(): void {
-    this.prepareSearchDenounce();
-    this.loadCountries();
-    this.loadFavourites();
-  }
-  private prepareSearchDenounce(): void {
-    this.searchBar.ionInput
-      .pipe(
-        debounceTime(300)
-      )
-      .subscribe(event => {
-        this.filterCountries((event.target as HTMLInputElement).value);
-      });
-  }
-
-  private loadCountries(): void {
-    this.countryService.getCountries().pipe(
-      catchError(error => this.handleError(error)),
-      tap(result => {
-        if (result && !result.error) {
-          this.countries = result.data;
-          this.baseCountries = result.data;
-          this.groupCountries();
-          this.createLetters();
-          this.baseBackupGroupedCountries = { ...this.groupedCountries };
-        }
-      })
-    ).subscribe();
+    Promise.all([this.prepareSearchDenounce(), this.loadCountries(), this.loadFavourites()]).catch((error) => {
+      this.handleError(error);
+    })
+  }
+  private prepareSearchDenounce(): Promise<void> {
```

Comment:
@mnehro use ionic build in debounce if you use ion-searchbar below link : 
https://ionicframework.com/docs/api/searchbar

Reply:
I already removed it and used this debounce property in ion-search bar
    <ion-searchbar #searchBar animated="true" placeholder="Search" (ionInput)="filterCountries($event.target.value)"
      debounce="300"></ion-searchbar>